sdk/core/src/services/github/client/mod.rs

```diff
@@ -0,0 +1,284 @@
+        *vector.last_mut().unwrap() = num_last_call;
+
+        Ok(vector)
+    }
+
+    /// Function to get the data for a page of repos
+    pub async fn get_page_of_repos(
+        &self,
+        org: &String,
+        page: usize,
+        per_page: &u32,
+    ) -> Result<Vec<Repo>, Error> {
+
+        let github_org_url = format!("{GH_URL}/orgs/{org}/repos?type=sources&page={page}&per_page={per_page}");
+
+        println!("Calling({})", github_org_url);
+
+        let response: Result<Option<Vec<Repo>>, HyperError> = get(
+            github_org_url.as_str(),
+            ContentType::Json,
+            self.token.as_str(),
+            None::<String>,
+        ).await;
+
+        match response {
+            Ok(option) => match option {
+                Some(value) => Ok(value),
+                None => Err(
+                    Error::GitHubErrorResponse(
+                        "Get request from GitHub had an empty response".to_string()
+                    )
+                ),
+            },
+            Err(err) => Err(
+                Error::GitHubResponse(err)
+            ),
+        }
+    }
+
+    /// Generates a unique clone path for a repository.
+    pub fn clone_path(&self, url: &str, hash: &String) -> String {
+
+        let repo_name = url
+            .split('/')
+            .collect::<Vec<&str>>()
+            .pop()
+            .unwrap()
+            .replace(".git", "");
+
+        format!("/harbor-output/{}/{}", hash, repo_name)
+    }
+
+    /// Clones a git repository to the specified clone path.
+    pub fn clone_repo(&self, clone_path: &str, url: &str) -> Result<(), Error> {
+
+        println!("PROCESSING> Cloning repo: {}", url);
+
+        match Repository::clone(url, clone_path) {
+            Err(err) => return Err(
+                Error::GitHubErrorResponse(
+                    format!("==> error cloning repository from {}: {}", url, err)
+                )
+            ),
+            _ => info!("Successfully cloned repo"),
+        };
+
+        Ok(())
+    }
+
+    /// Removes a cloned repository from the filesystem.
+    pub fn remove_clone(&self, clone_path: &str) -> std::io::Result<()> {
+        if Path::new(&clone_path).is_dir() {
+            return std::fs::remove_dir_all(clone_path);
+        }
+
+        Ok(())
+    }
+}
+
+/// Commit represents the returned Json from a commits
+/// request from the GitHub API.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Commit {
+    /// Commit hash of main branch in repository
+    pub sha: Option<String>
+}
+
+/// Org is used to extract the number of Public Repos
+/// in a given GitHub Organization.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Org {
+    /// The number of Public Repos in
+    /// this organization
+    public_repos: Option<u32>
+}
+
+/// Repo is used to extract several values from a Request for
+/// the Repositories in a given GitHub Organization
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Repo {
```

Comment:
Collaboration point.  I love that you created types specific to the client module here, so that we can keep them separate from the Harbor entities. One thing that I think would be really good to add to the Harbor `Repository` entity is a `Vec<Commit>` or something like that, so we can keep a timestamped history of commit hashes at time of inspection/sync instead of a single value/field on the entity.  Food for thought as we work on the final solution together.

Reply:
Yes, that sounds great.  I've added this task to my personal backlog